scipy/stats/tests/test_distributions.py

```diff
@@ -3450,6 +3450,18 @@ def test_burr12_ppf_small_arg():
     #   5.7735026918962575e-09
     assert_allclose(quantile, 5.7735026918962575e-09)
 
+def test_burr_nan_mean_var():
+    c, d = 0.5, 3
+    mean, variance = stats.burr(c, d).stats()
+    assert_(np.isnan(mean))
+    assert_(np.isnan(variance))
+
+def test_burr_stats():
+    c, d = 5.0, 3
+    mean, variance = stats.burr(c, d).stats()
+    mean_hc, variance_hc = 1.4110263183925857, 0.22879948026191643
```

Comment:
An obligatory question about hardcoded numbers: how to reproduce them? IOW, if this test starts failing in the future, how does a future maintainer check these numbers?

Reply:
added a comment, these should come from the formula for mean and variance: 

https://edoc.unibas.ch/61230/1/20180305141821_5a9d439d76be5.pdf

page 6 or 8 depending on if you use pdf viewer page numbering or the numbering written on each page of the document. Eqn (7) gives the Kth moment. In the `burr_gen` class `b=1` and `p` in the pdf corresponds to `d` here in the code. The `a` in the pdf corresponds to the `c` here in the code. I'll double check these calculations tomorrow evening.

Reply:
For the `mean_hc` and `variance_hc` I get:
```
>>> sc.beta(3 + 1/5, 1. - 1/5) * 3 
1.4110263183925855
>>> sc.beta(3 + 2/5, 1. - 2/5) * 3  - (sc.beta(3 + 1/5, 1. - 1/5) * 3)**2
0.2287994802619171
```

note there might be some loss in precision in the last few digits displayed. 
Which corresponds to what I would expect based on formulas in the manuscript:

https://edoc.unibas.ch/61230/1/20180305141821_5a9d439d76be5.pdf

and agrees with the output of `_stats` e.g. 

```

```